testutil/ciao-down/ciao_down.go

```diff
@@ -460,6 +460,7 @@ func status(ctx context.Context, errCh chan error) {
 }
 
 func connect(ctx context.Context, errCh chan error) {
+	timeout := time.NewTimer(time.Second * 30)
```

Comment:
You could use the existing context to do the time out, e.g.

```
ctx, cancel = context.WithTimeout(ctx, time.Second * 30)
defer cancel()
```

Then you don't need to modify the select statement at all.  If you wanted to distinguish between a cancel and a timeout you could compare ctx.Error() to context. DeadlineExceeded

Reply:
I like this idea.